pithos/plugins/notify.py

```diff
@@ -35,40 +35,36 @@ def on_prepare(self):
             os.environ['GNOTIFICATION_BACKEND'] = 'freedesktop'
 
         self._app = Gio.Application.get_default()
+        self._app_id = self._app.get_application_id()
+        self._fallback_icon = Gio.ThemedIcon.new('audio-x-generic')
         self.prepare_complete()
 
     def on_enable(self):
-        self.song_change_handler = self.window.connect('song-changed', self.send_notification)
-        self.state_change_handler = self.window.connect('user-changed-play-state', self.send_notification)
+        self._song_change_handler = self.window.connect('song-changed', self.send_notification)
 
     def send_notification(self, window, *ignore):
         if window.is_active():
-            return
-
-        song = window.current_song
-        # This matches GNOME-Shell's format
-        notification = Gio.Notification.new(song.artist)
-        notification.set_body(song.title)
-        if song.artUrl:
-            notification.set_icon(Gio.FileIcon.new(Gio.File.new_for_uri(song.artUrl)))
-
-        if window.playing:
-            notification.add_button(_('Pause'), 'app.pause')
+            # GNOME-Shell will auto dismiss notifications
+            # when the window becomes "active" but other DE's may not (KDE for example).
+            # If we're not going to replace a previous notification
+            # we should withdraw said stale previous notification.
+            self._app.withdraw_notification(self._app_id)
```

Comment:
This seems like an odd place to do this. If you want the notification to go away with window focus then you should connect to window focus?

Reply:
It's not to make notifications go away when the widow focuses it's to dismiss notifications from previous songs if we're not going to replace them. GNOME Shell just happens to auto dismiss notifications when the app is "active".

Reply:
We probably also want to withdraw any notifications on shutdown.

Reply:
Something like:

```python
    def do_shutdown(self):
        self.withdraw_notification(self.get_application_id())
        Gtk.Application.do_shutdown(self)
        if self.window:
            self.window.destroy()
```

Reply:
Yea, the plugin can connect to the shutdown signal.

Reply:
Again GNOME will auto dismiss notifications when we shutdown (with the freedesktop backend) KDE does not.